test/run.js

```diff
@@ -0,0 +1,27 @@
+const test = require("node:test");
+const path = require("path");
+const fs = require("fs");
+
+// don't want to recursively run this file
+const excluded_test_files = ["run.js", "package.json", "utils.js"];
+// should be run from the repo root dir
+const testDir = path.join(process.cwd(), 'test');
+
+// recursively get all files from a folder
+// may be overkill since we only expect 2 child folders in
+// 'unit' and 'integration' but it couldn't hurt
+function get_files(dirName) {
+    var files = fs.readdirSync(dirName);
+    result = [];
+    files.forEach((file) => {
+        var filepath = path.join(dirName, file);
+        if (fs.statSync(filepath).isDirectory()) {
+            result.concat(get_files(filepath));
+        } else if (!excluded_test_files.includes(file)) {
+            result.push(filepath);
+        }
+    });
+    return result;
+}
+
+test.run({files: get_files(testDir)}).pipe(process.stdout);
```

Comment:
Should we pipe an output file instead of stdout? I think either option is fine, just wanted to bring it up as a possibility.

Reply:
We can see all pervious runs on the actions page, and each one's standard output, so it should be saved. We can definitely do this but it should be saved already if we're worried about viewing the details of the runs

Reply:
Sounds good to me!